### PR TITLE
Redirect test for ATS

### PR DESCRIPTION
**Pull request by macisasandwich**

### CODE REVIEW

**Comment by dragon512** [comment]
[approve ci]

**Comment by persiaAziz** [comment]
[approve ci]

File: tests/gold_tests/redirect/gold/redirect.gold [hunk 1]
```diff
@@ -0,0 +1,29 @@
+HTTP/1.1 200 OK
+Server: ATS/``
+Date: ``
+Content-Type: application/json
+Access-Control-Allow-Origin: *
+Access-Control-Allow-Credentials: true
+X-Powered-By: Flask
+X-Processed-Time: ``
+Content-Length: ``
+Via: 1.1 vegur
```
Comment (persiaAziz) [inline tests/gold_tests/redirect/gold/redirect.gold]: Change this to Via:``?

**Comment by persiaAziz** [comment]
[approve ci]

File: tests/gold_tests/redirect/redirect.test.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ts = Test.MakeATSProcess("ts")
+
+ts.Disk.records_config.update({
+	'proxy.config.http.redirection_enabled': 1,
+	'proxy.config.http.number_of_redirections': MAX_REDIRECT
+})
+
+# TODO use httpbin's redirect X times link
+ts.Disk.remap_config.AddLine(
+	'map http://127.0.0.1:{0} http://httpbin.org/redirect-to?url=http://httpbin.org/anything'.format(ts.Variables.port)
```
Comment (persiaAziz) [inline tests/gold_tests/redirect/redirect.test.py]: I agree with @scw00 , you might want to replace httpbin with microserver here.

Comment (dragon512) [inline tests/gold_tests/redirect/redirect.test.py]: I don't think we should be using public IPs. These may not be accessible

**Comment by dragon512** [comment]
[approve ci]

**Pull request merged**